loki/transformations/utilities.py

```diff
@@ -623,20 +639,94 @@ def find_driver_loops(section, targets):
         List of subroutines that are to be considered as part of
         the transformation call tree.
     """
+    class FindDriverLoops(Visitor):
+        """
+        A  visitor that collects all driver loops in a section of the IR.
+
+        If there are nested loops, then the highest level pragma-marked driver
+        loop will take precedence and be considered the driver loop. If there
+        is no pragma-marked driver loop, then the highest level loop that does
+        not contain a driver loop will be considered the driver loop.
+        """
+
+        def __init__(self):
+            super().__init__()
+            self.driver_loops = []
 
-    driver_loops = []
-    nested_driver_loops = []
-    for loop in FindNodes(ir.Loop).visit(section):
-        if loop in nested_driver_loops:
-            continue
+        def visit_tuple(self, o, **kwargs):
+            nested_pragma_loop = False
+            for i in o:
+                nested_pragma_loop |= self.visit(i, **kwargs)
+            return nested_pragma_loop
 
-        if not is_driver_loop(loop, targets):
-            continue
+        visit_list = visit_tuple
+
+        def visit_CallStatement(self, call, **kwargs):
+            targets = kwargs.get('targets', [])
+            has_target_calls = kwargs.get('has_target_calls', None)
+            if has_target_calls is not None and call.name in targets:
+                has_target_calls.append(True)
+            return False
```

Comment:
With the above addition to ensure `targets` is a tuple, you can use it here directly and return the triplet as suggested:
```suggestion
        def visit_CallStatement(self, call, **kwargs):
            if call.name in targets:
                return False, [], [call]
            return self.default_retval()
```